### DISCUSSION TITLE
transport for walletConnect

### DISCUSSION THREAD

**Discussion by t9fiction** [wevm/viem#879] [Question] [answered]
Hi, I am trying to use the walletClient for writing a contract function, So it works fine with the `transport: custom(window.ethereum)`
but that on the desktop/laptops as they have metamask installed, but as soon as i try to open it in the mobile, it shows blank screen as there is no window.ethereum in mobile browsers.
browers like brave and kiwi works fine though as they have builtin wallets, so how to use it in mobile, like 
what should be used as transport in walletConnect that can fullfill both mobile and desktop versions

**Comment by jxom** [comment, marked as answer]
Here is an example of using WalletConnect: https://stackblitz.com/edit/viem-getting-started-a3ekcn?file=index.tsx

**Reply by OoXooOx** [reply]
But what should I do when I use web3modal? In their docs said, that *Note: A new session is created automatically when the modal is opened, so avoid calling provider.connect by yourself." ? And even when I implement all code above in my project, when I click connect I have pop up window with differ wallets for connect. There is "Ledger", "Zerion". There is no "Metamask" , "Mew".  How I can fetch transport for web3modal and send it to createWalletClient from viem?

**Reply by hcwilk** [reply]
Is there any follow up on this? I'm running into the same issue. I know I can manually implement each provider and have each connection process be separate, but I thought the point of Viem was to have all of this be consolidated into the Web3ModalFlow. 

I can call all my readContracts totally file, but how can I use the Web3Modal flow to run transactions with the WalletConnect provider?

**Reply by codekoriko** [reply]
> Here is an example of using WalletConnect: https://stackblitz.com/edit/viem-getting-started-a3ekcn?file=index.tsx

same here I got the example to work, just to realized that the set of wallet offers by the Modal is all obscur/unknown wallet, no metamask for example. 

Why is the list differs from the one offers in the [Web3Inbox](https://app.web3inbox.com/login?next=%2Fnotifications) for example? 

Is because it's use an older protocol version?

**Comment by ayaamdouni** [comment]
Hello @t9fiction, I have the same problem. Have you found any solution to use walletConnect with `transport: custom(window.ethereum)`

Thank you

**Reply by jxom** [reply]
it's here: https://github.com/wevm/viem/discussions/879#discussioncomment-6422248
